tests/unit/package_managers/test_npm.py

```diff
@@ -897,7 +1025,6 @@ def test_resolve_npm_validation(
                     "bar": {
                         "version": "2.0.0",
                         "resolved": "https://registry.npmjs.org/bar/-/bar-2.0.0.tgz",
-                        "integrity": "sha512-JCB8C6SnDoQf",
```

Comment:
Why is `"missing_hash_in_file": None` for this test even though the integrity is missing?

Reply:
I thought it's the correct behavior since it's not missing in `node_modules/bar` [here](https://github.com/containerbuildsystem/cachi2/pull/301/files/95fa45da3379071b12ad11e5446874a490ad8d8b#diff-628fe82bed25383be81718c4bf663b93c2aad092de52f56b3bb9303f24197391L893).

Reply:
Ah, right, yeah that's correct. If that's intentional (having integrity in `packages` but not in `dependencies`), could you add a comment about that?

Reply:
yup, I did it on purpose, added a comment

Reply:
If the opposite happened (integrity is present in the `dependencies` key, but missing from `packages`), then we would still report it as missing, right? IIUC, we are completely ignoring the dependencies key in the PackageLock class.

Do we want to change one of the test cases here to cover this scenario as well?

Reply:
Good point Bruno, I believe this scenario is worth it. +1